subarray.go

```diff
@@ -52,6 +52,101 @@ func (sa *Subarray) Free() {
 	}
 }
 
+// SetSubarray sets a subarray, defined in the order dimensions were added.
+// Coordinates are inclusive. For the case of writes, this is meaningful only
+// for dense arrays, and specifically dense writes.
+func (sa *Subarray) SetSubArray(subArray interface{}) error {
+
+	if reflect.TypeOf(subArray).Kind() != reflect.Slice {
+		return fmt.Errorf("Subarray passed must be a slice, type passed was: %s", reflect.TypeOf(subArray).Kind().String())
+	}
+
+	subArrayType := reflect.TypeOf(subArray).Elem().Kind()
+
+	schema, err := sa.array.Schema()
+	if err != nil {
+		return fmt.Errorf("Could not get array schema from query array: %s", err)
```

Comment:
We may have to update the error message because the method no longer operates on queries, no?